packages/zoe/test/unitTests/contracts/vpool-xyk-amm/test-xyk-amm-swap.js

```diff
@@ -0,0 +1,1100 @@
+// @ts-check
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { test } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+
+import path from 'path';
+
+import bundleSource from '@agoric/bundle-source';
+import { makeIssuerKit, AmountMath } from '@agoric/ertp';
+import { E } from '@agoric/eventual-send';
+import { assert, q } from '@agoric/assert';
+import fakeVatAdmin from '../../../../tools/fakeVatAdmin.js';
+
+// noinspection ES6PreferShortImport
+import { makeZoeKit } from '../../../../src/zoeService/zoe.js';
+import { setup } from '../../setupBasicMints.js';
+import {
+  makeTrader,
+  updatePoolState,
+  priceFromTargetOutput,
+  outputFromInputPrice,
+} from '../../../autoswapJig.js';
+import buildManualTimer from '../../../../tools/manualTimer.js';
+import {
+  getAmountOut,
+  makeRatio,
+  ceilMultiplyBy,
+  natSafeMath,
+} from '../../../../src/contractSupport/index.js';
+import {
+  assertAmountsEqual,
+  assertPayoutAmount,
+} from '../../../zoeTestHelpers.js';
+
+const { ceilDivide } = natSafeMath;
+
+const filename = new URL(import.meta.url).pathname;
+const dirname = path.dirname(filename);
+
+const ammRoot = `${dirname}/../../../../src/contracts/vpool-xyk-amm/multipoolMarketMaker.js`;
+
+test('amm with valid offers', async t => {
+  const { moolaR, simoleanR, moola, simoleans } = setup();
+  const { zoeService } = makeZoeKit(fakeVatAdmin);
+  const feePurse = E(zoeService).makeFeePurse();
+  const zoe = E(zoeService).bindDefaultFeePurse(feePurse);
+  const invitationIssuer = await E(zoe).getInvitationIssuer();
+  const invitationBrand = await E(invitationIssuer).getBrand();
+
+  // Set up central token
+  const centralR = makeIssuerKit('central');
+  const centralTokens = value => AmountMath.make(value, centralR.brand);
+
+  // Setup Alice
+  const aliceMoolaPayment = moolaR.mint.mintPayment(moola(100000));
+  // Let's assume that central tokens are worth 2x as much as moola
+  const aliceCentralPayment = centralR.mint.mintPayment(centralTokens(50000));
```

Comment:
If I refine the parameter type of `centralTokens` from `any` to `Value`, I get:

```
Argument of type 'number' is not assignable to parameter of type 'Value'.
```

Does it seem cost-effective to change these?

```suggestion
  const aliceCentralPayment = centralR.mint.mintPayment(centralTokens(50000n));
```

_p.s. postponed to #3914_